examples/hyperparameter_tuning/steps/evaluation/compare_scores.py

```diff
@@ -0,0 +1,31 @@
+from dynamic_pipelines.gather_step import GatherStepsParameters
+from steps.evaluation.evaluation_parameters import EvaluationOutputParams
+
+from zenml.steps import step
+
+
+class CompareScoreParams(GatherStepsParameters):
+    """Parameters for compare_score step"""
+
+    reduce_min: bool = False
+    reduce_max: bool = False
+
+
+@step
+def compare_score(params: CompareScoreParams) -> dict:
+    """Compare scores over multiple evaluation outputs."""
+    outputs = EvaluationOutputParams.gather(params)
+
+    if params.reduce_min:
+        output = min(outputs, key=lambda x: x.score)
+        print(
+            f"minimal value at {output.model_parameters}. score = {output.score*100:.2f}%"
+        )
+        return output.model_parameters
+
+    if params.reduce_max:
+        output = max(outputs, key=lambda x: x.score)
+        print(
+            f"maximal value at {output.model_parameters}. score = {output.score*100:.2f}%"
+        )
+        return output.model_parameters
```

Comment:
Maybe throw an error here if both `reduce_min` and `reduce_max` are `False`?

Reply:
That's a good idea. Perhaps, that can be achieved by adding a [`pydantic` validator](https://docs.pydantic.dev/usage/validators/) to the class.

Reply:
I'll think of a safer implementation for this. Thanks.